kamu-adapter-http/tests/tests/test_smart_push.rs

```diff
@@ -0,0 +1,256 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::str::FromStr;
+
+use kamu::{
+    domain::*,
+    testing::{DatasetTestHelper, MetadataFactory},
+};
+use opendatafabric::*;
+
+use crate::harness::{
+    await_client_server_flow, copy_folder_recursively, ClientSideHarness, ServerSideHarness,
+};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[test_log::test(tokio::test)]
+#[cfg_attr(feature = "skip_docker_tests", ignore)]
```

Comment:
Just a thought: 

We probably could run the majority of these tests dockerless, using in-memory storage and having server issue proxied (simple transfer protocol) URLs for object download/upload.

Then we could have just a couple of tests that do use docker for specifically testing the delegated / non-proxied URLs with Minio.